test/extended/authorization/scc.go

```diff
@@ -163,4 +163,49 @@ var _ = g.Describe("[sig-auth][Feature:PodSecurity][Feature:SCC]", func() {
 		o.Expect(err).NotTo(o.HaveOccurred())
 		o.Expect(pod.Annotations[securityv1.ValidatedSCCAnnotation]).To(o.Equal("restricted-v2"))
 	})
+
+	g.It("SCC admission fails for incorrect/non-existent required-scc annotation", func() {
+		sccPod := &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "required-scc-testpod",
+				Annotations: map[string]string{
+					securityv1.RequiredSCCAnnotation: "non_existent_scc", // Set an annotation to a non-existent SCC.
+				},
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:    "sleeper",
+						Image:   "fedora:latest",
+						Command: []string{"sleep"},
+						Args:    []string{"infinity"},
+						SecurityContext: &corev1.SecurityContext{
+							RunAsNonRoot: pointer.Bool(true),
+							Capabilities: &corev1.Capabilities{
+								Drop: []corev1.Capability{
+									"ALL",
+								},
+							},
+						},
+					},
+				},
+				SecurityContext: &corev1.PodSecurityContext{
+					RunAsNonRoot: pointer.Bool(true),
+				},
+			},
+		}
+
+		// Attempt to create the pod and expect an error due to the non-existent SCC.
+		_, err := oc.KubeClient().CoreV1().Pods(oc.Namespace()).Create(context.Background(), sccPod, metav1.CreateOptions{})
+		o.Expect(err).To(o.HaveOccurred())
+		o.Expect(err.Error()).To(o.MatchRegexp(`required .*non_existent_scc.* not found`))
```

Comment:
Per https://github.com/openshift/apiserver-library-go/pull/119/files#diff-a23469582da72fa287a9f4cc403e4c6a9f031648d0ac95246bb926a7e1703374R237 `fmt.Errorf("failed to retrieve the required SCC %q: %w", requiredSCCName, err)`, I'm afraid the message may not match `required .*non_existent_scc.* not found`?

Reply:
I attempted to run this test on a pre-merged [1661](https://github.com/openshift/kubernetes/pull/1661) cluster. The following below error message was generated by the cluster in response of the failure. During test execution, the expression above matched (Test Passed) because anything in between is ignored by `regexp` `.*` .

```
<string>: pods "required-scc-testpod" is forbidden: failed to retrieve the required SCC "non_existent_scc": securitycontextconstraints.security.openshift.io "non_existent_scc" not found
to match regular expression
```
Also will execute this test again after PR merged [1661](https://github.com/openshift/kubernetes/pull/1661)